source/blender/io/ply/importer/ply_import.cc

```diff
@@ -152,6 +156,11 @@ void importer_main(Main *bmain,
     else if (words[0] == "end_header") {
       break;
     }
+    else if ((words[0][0] >= '0' && words[0][0] <= '9') || words[0][0] == '-') {
```

Comment:
I'm pretty sure this only works on ASCII formatted files.

I think a better alternative is to just keep reading lines until EOF and if EOF is encountered before end_header it means the header wasn't terminated properly.